com.unity.ml-agents/Runtime/SideChannels/SideChannel.cs

```diff
@@ -34,9 +35,18 @@ public Guid ChannelId
 
         internal void ProcessMessage(byte[] msg)
         {
-            using (var incomingMsg = new IncomingMessage(msg))
+            try
```

Comment:
This threw me off when I was working on the Training Analytics side channel. If the SideChannel throws here, we disconnect the trainer. In general, I think we should wrap user code in bubble wrap.